From 159116bf409a6318b317d4746aac73867d167cf0 Mon Sep 17 00:00:00 2001
From: Li Jun <b47624@freescale.com>
Date: Thu, 18 Sep 2014 22:59:40 +0800
Subject: [PATCH] MLK-9618-2 usb: chipidea: otg: update is_b_host flag when
 setup&stop host

This patch sets and clear is_b_host flag of usb_bus when start and stop host in
chipidea host driver.

Acked-by: Peter Chen <peter.chen@freescale.com>
Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit c164e65922235efcacc3743a9719adc7a9ef7413)
---
 drivers/usb/chipidea/host.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/usb/chipidea/host.c b/drivers/usb/chipidea/host.c
index 5a87438..2d9307d 100644
--- a/drivers/usb/chipidea/host.c
+++ b/drivers/usb/chipidea/host.c
@@ -266,6 +266,13 @@ static int host_start(struct ci_hdrc *ci)
 		}
 	}
 
+	if (ci_otg_is_fsm_mode(ci)) {
+		if (ci->fsm.id && ci->fsm.otg->state <= OTG_STATE_B_HOST)
+			hcd->self.is_b_host = 1;
+		else
+			hcd->self.is_b_host = 0;
+	}
+
 	ret = usb_add_hcd(hcd, 0, 0);
 	if (ret) {
 		goto disable_reg;
@@ -316,6 +323,8 @@ static void host_stop(struct ci_hdrc *ci)
 		if (ci->platdata->reg_vbus && !ci_otg_is_fsm_mode(ci) &&
 			(ci->platdata->flags & CI_HDRC_IMX_VBUS_EARLY_ON))
 				regulator_disable(ci->platdata->reg_vbus);
+		if (hcd->self.is_b_host)
+			hcd->self.is_b_host = 0;
 	}
 	ci->hcd = NULL;
 }
-- 
1.8.0

